### PR TITLE
Clean up testing

### PR DESCRIPTION
**Pull request by lugray**
commit 8e5713c2ae01f26e608350dfbf40f3a30860073c (HEAD -> test_improvements, origin/test_improvements)
Author: Lisa Ugray <lisa.ugray@shopify.com>
Date:   Sat Mar 16 01:43:01 2019 -0400

    Add travis

commit ee7db6ee85897a99a1e3e32057b8e33fde75f29e
Author: Lisa Ugray <lisa.ugray@shopify.com>
Date:   Sat Mar 16 01:02:54 2019 -0400

    Add timeout

    If we encounter an error that prevents getting back to a prompt, timeout
    and print output for debugging.

commit 82e2b39b571f0be6937b4ee8a23e77e92946c47c
Author: Lisa Ugray <lisa.ugray@shopify.com>
Date:   Sat Mar 16 00:13:09 2019 -0400

    Make tests more cross platform

    wc has padding to right align output that differs platform to platform.

commit c16cc3d7ca8305288c7961fedfebcda258aa16f3
Author: Lisa Ugray <lisa.ugray@shopify.com>
Date:   Fri Mar 15 23:51:47 2019 -0400

    Detect shells

    Detect all veersions of shells on PATH. Makes this more cross platform,
    and resilient to locations.

commit e32f8d6e704394f19833de95cfd9fb4c9b464582
Author: Lisa Ugray <lisa.ugray@shopify.com>
Date:   Fri Mar 15 23:39:53 2019 -0400

    Remove marker commands

    Remove commands used only as markers, and structure related input and
    output instead.

commit 4f2fca0656fe60079f8f013a28af1ff6a38061d3
Author: Lisa Ugray <lisa.ugray@shopify.com>
Date:   Fri Mar 15 22:54:01 2019 -0400

    Clean up zsh

    Invoke zsh with arguments that make the output sanitization
    unneccessary.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+language: ruby
+rvm: 2.4.1
+git:
+  depth: 1
+before_install:
+  - sudo add-apt-repository -y ppa:hyperair/zsh-backports # Trusty's zsh deos not support pipefail
```
Comment (lavoiesl) [inline .travis.yml:6]:
```suggestion
  - sudo add-apt-repository -y ppa:hyperair/zsh-backports # Trusty's zsh does not support pipefail
```

Comment (lavoiesl) [reply .travis.yml:6]: Doesn't that mean that our script should not rely on pipefail?

File: shell.rb [hunk 2]
```diff
@@ -1,33 +1,55 @@
 require('open3')
 require('pty')
+require('pathname')
+require('timeout')
 
 class Shell
+  def self.available_versions(shell)
+    ENV['PATH'].split(':').map do |d|
```
Comment (lavoiesl) [inline shell.rb]: I fix this incredibly hard to read.
How about:

```suggestion
    ENV['PATH'].split(':')
      .map { |d| File.join(d, shell) }
      .select { |f| File.file?(f) && File.executable?(f) }
      ...
```

It's also the recommended style:
> Prefer {...} over do...end for single-line blocks. Avoid using {...} for multi-line blocks. Always use do...end for “control flow” and “method definitions” (e.g. in Rakefiles and certain DSLs). Avoid do...end when chaining.

https://shopify.github.io/ruby-style-guide/

**Review by lavoiesl — COMMENTED** [review]
I don't quite understand what's going on or what is the purpose of the changes, can you add a PR description?

**Review by donk-shopify — APPROVED** [review]

File: .travis.yml (hunk 1 above)
Comment (lugray) [reply .travis.yml:6]: No - we're adding pipefail only for the testing to ensure that our script is resilient to users using that option.

**Review by lavoiesl — APPROVED** [review]

**Pull request merged**